### PR TITLE
Add documentation to Time class

### PR DESCRIPTION
**Pull request by adlerhsieh**
Add introductory texts to Time class by bringing up several commonly used methods

### CODE REVIEW

File: src/time/time.cr [hunk 1]
```diff
@@ -1,3 +1,50 @@
+# Time library allows you to inspect, analyze, and calculate time. Here are some examples:
```
Comment (jhass) [inline src/time/time.cr]: "The Time library", "format time" too ;) and maybe we should mention time spans (`Time::Span`) too?

File: src/time/time.cr [hunk 2]
```diff
@@ -1,3 +1,50 @@
...
+#     time.day     #=> 15
+#     time.hour    #=> 10
+#     time.minute  #=> 20
+#     time.second  #=> 30
+# 
+#     # Creating a time instance
+#     Time.new(2016,2,15)
+#     #=> 2016-02-15 00:00:00
+# 
+#     # Specifying time & timezone
```
Comment (jhass) [inline src/time/time.cr]: Well, not exactly a timezone yet, just UTC vs localtime, though the latter is kinda broken, see #1521

File: src/time/time.cr [hunk 3]
```diff
@@ -1,3 +1,50 @@
...
+#     Time.new(2016,2,15,10,20,30,Time::Kind::Utc)
+#     #=> 2016-02-15 10:20:30 UTC 
+#     
+# ### Formatting Time
+# 
+# `to_s` method returns a String value in assigned format.
```
Comment (jhass) [inline src/time/time.cr]: "The" ;)

File: src/time/time.cr [hunk 4]
```diff
@@ -1,3 +1,50 @@
...
+# 
+#     Time.now.to_s("%Y-%m-%d")
+#     #=> "2015-10-12"
+#     
+#     # Arguments include but are not limited to:
```
Comment (jhass) [inline src/time/time.cr]: Let's call them format specifiers.

File: src/time/time.cr [hunk 5]
```diff
@@ -1,3 +1,50 @@
...
+#     %S  => second
+#     %D  => date
+#     %u  => weekday
+#     
+# ### Calculation
+# 
+#     Time.new(2015,10,10) - 5.day
+#     #=> 2015-10-05 00:00:00
+#     
+#     (Time.new(2015,10,10) - Time.new(2015,9,10)).days
```
Comment (jhass) [inline src/time/time.cr]: Maybe reference `Time::Span` here?

**Comment by adlerhsieh** [comment]
Docs updated. I've removed to the timezone part.

File: src/time/time.cr [hunk 6]
```diff
@@ -1,3 +1,63 @@
...
+#     
+#     time.year    #=> 2015
+#     time.month   #=> 2
+#     time.day     #=> 15
+#     time.hour    #=> 10
+#     time.minute  #=> 20
+#     time.second  #=> 30
+#     time.monday? #=> true
+#
+#     # Creating a time instance with date only
```
Comment (jhass) [inline src/time/time.cr]: Uh, missed a couple last time, sorry. "a date" ;)

File: src/time/time.cr [hunk 7]
```diff
@@ -1,3 +1,63 @@
...
+#     Time.new(2016,2,15)
+#     #=> 2016-02-15 00:00:00
+# 
+#     # Specifying time
```
Comment (jhass) [inline src/time/time.cr]: "a time"

File: src/time/time.cr [hunk 8]
```diff
@@ -1,3 +1,63 @@
...
+#     Time.new(2016,2,15,10,20,30)
+#     #=> 2016-02-15 10:20:30 UTC 
+#     
+# ### Formatting Time
+# 
+# The `to_s` method returns a String value in the assigned format.
```
Comment (jhass) [inline src/time/time.cr]: Maybe link "\`String\`"?

File: src/time/time.cr [hunk 9]
```diff
@@ -1,3 +1,63 @@
...
+#     %D  => date
+#     %u  => weekday
+#     
+# ### Calculation
+# 
+#     Time.new(2015,10,10) - 5.day
+#     #=> 2015-10-05 00:00:00
+#     
+#     # Time calculation returns a Time::Span instance,
+#     # which can be analyzed with its method.
```
Comment (jhass) [inline src/time/time.cr]: "methods"

File: src/time/time.cr [hunk 10]
```diff
@@ -1,3 +1,63 @@
...
+#     span = Time.new(2015,10,10) - Time.new(2015,9,10)
+#     span.days          #=> 30
+#     span.total_hours   #=> 720     
+#     span.total_minutes #=> 43200
+#     
+#     # Calculation bewteen Time::Span instances
```
Comment (jhass) [inline src/time/time.cr]: "between"

**Comment by adlerhsieh** [comment]
Ah, so many grammatical mistakes :P Docs updated. Thanks!

File: src/time/time.cr [hunk 11]
```diff
@@ -1,3 +1,63 @@
...
+#     Time.new(2016,2,15)
+#     #=> 2016-02-15 00:00:00
+# 
+#     # Specifying a time
+#     Time.new(2016,2,15,10,20,30)
+#     #=> 2016-02-15 10:20:30 UTC 
+#     
+# ### Formatting Time
+# 
+# The `to_s` method returns a [String](https://github.com/manastech/crystal/blob/master/src/string.cr) value in the assigned format.
```
Comment (jhass) [inline src/time/time.cr]: Ah sorry, I meant literally "`String`". The doc parser will link it for you.

Comment (adlerhsieh) [inline src/time/time.cr]: Yeap, sure, I did not know the parser does it for us. Changing it to `String`.

Comment (adlerhsieh) [inline src/time/time.cr]: Updated :)

**Comment by jhass** [comment]
Thank you!

**Pull request merged**